### ISSUE TITLE
DPO训练模型输出 UserWarning: None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by weiliang987644015** [modelscope/ms-swift#2298]
**Describe the bug**
DPO训练模型，输出警告是否对训练产生影响？
`UserWarning: None of the inputs have requires_grad=True. Gradients will be None`
![swift_bug_2](https://github.com/user-attachments/assets/43f8a54f-895c-41ac-a177-23106d376f4f)

**Your hardware and system info**
ms-swift==2.6.0dev0
pytorch==2.4.0+cu121
python==3.11.10
CUDA==12.1
trl==0.11.4
transformers==4.45.2
GPU：H800-80G

**Comment by Jintao-Huang** [comment]
有shell吗, 我复现一下

**Comment by weiliang987644015** [comment]
> 有shell吗, 我复现一下

shell脚本：
![image](https://github.com/user-attachments/assets/2e8d9579-6547-4214-938a-a1ea32281291)

**Comment by Jintao-Huang** [comment]
不用管的 ref_model forward 不计算梯度的.

**Comment by weiliang987644015** [comment]
> 不用管的 ref_model forward 不计算梯度的.

好的，感谢解答。
